go-controller/pkg/node/gateway_udn.go

```diff
@@ -717,3 +767,83 @@ func addRPFilterLooseModeForManagementPort(mgmtPortName string) error {
+		for netName, netConfig := range udng.openflowManager.defaultBridge.netConfig {
+			if netName == networkName {
+				netConfig.isUDNNetworkAdvertised = isUDNNetworkAdvertised
+			} else if netName == types.DefaultNetworkName {
+				netConfig.isUDNNetworkAdvertised = udng.isPodNetworkAdvertised
+			}
+		}
+	}
+}
+
+func (udng *UserDefinedNetworkGateway) Reconcile() error {
+	klog.Info("Reconciling UDN gateway with updates")
+	if err := udng.updateUDNVRFIPRule(); err != nil {
+		return fmt.Errorf("error while updating ip rule for UDN %s: %s", udng.GetNetworkName(), err)
+	}
+
+	if err := udng.updateUDNFlow(); err != nil {
+		return fmt.Errorf("error while updating logical flow for UDN %s: %s", udng.GetNetworkName(), err)
+	}
+
+	return nil
+}
+
+// updateUDNVRFIPRule modifies existing IP rule for an UDN nework when that particular
+// network(10.132.0.0/14) is advertised:
+// Existing IP rule after creation of UDN network:
+// 2000:	from all fwmark 0x1001 lookup 1009
+// 2000:	from all to 169.254.0.12 lookup 1009
+// IP rule after the network is advertised:
+// 2000:	from all fwmark 0x1001 lookup 1009
+// 2000:	from all to 10.132.0.0/14 lookup 1009
+func (udng *UserDefinedNetworkGateway) updateUDNVRFIPRule() error {
+	interfaceName := util.GetNetworkScopedK8sMgmtHostIntfName(uint(udng.networkID))
+	mplink, err := util.LinkByName(interfaceName)
+	if err != nil {
+		return fmt.Errorf("unable to get link for %s, error: %v", interfaceName, err)
+	}
+	vrfTableId := util.CalculateRouteTableID(mplink.Attrs().Index)
+
+	if err = udng.ruleManager.DeleteWithMetadata(udng.GetNetworkRuleMetadata()); err != nil {
+		return fmt.Errorf("unable to delete iprule for network %s, err: %v", udng.GetNetworkName(), err)
+	}
+	udnReplyIPRules, err := udng.constructUDNVRFIPRules(vrfTableId)
+	if err != nil {
+		return fmt.Errorf("unable to get iprules for network %s, err: %v", udng.GetNetworkName(), err)
+	}
+
+	for _, rule := range udnReplyIPRules {
+		if err = udng.ruleManager.AddWithMetadata(rule, udng.GetNetworkRuleMetadata()); err != nil {
+			return fmt.Errorf("unable to create iprule %v for network %s, err: %v", rule, udng.GetNetworkName(), err)
+		}
+	}
+	return nil
+}
+
+// updateUDNFlow adds below OpenFlows based on the gateway mode and whether the network
+// is advertised or not:
+// table=1, n_packets=0, n_bytes=0, priority=16,ip,nw_dst=128.192.0.2 actions=LOCAL (Both gateway modes)
+// table=1, n_packets=0, n_bytes=0, priority=15,ip,nw_dst=128.192.0.0/14 actions=output:3 (shared gateway mode)
+func (udng *UserDefinedNetworkGateway) updateUDNFlow() error {
+	node, err := udng.watchFactory.GetNode(udng.nodeIPManager.nodeName)
+	if err != nil {
+		return fmt.Errorf("unable to get node %s: %s", udng.nodeIPManager.nodeName, err)
+	}
+	subnets, err := util.ParseNodeHostSubnetAnnotation(node, types.DefaultNetworkName)
+	if err != nil {
+		return fmt.Errorf("failed to get subnets for node: %s for OpenFlow cache update; err: %w", node.Name, err)
+	}
```

Comment:
We should not need to pass the subnets if they are already stored in `openflowManager.defaultBridge.netConfig`